src/vs/workbench/contrib/accountEntitlements/browser/accountsEntitlements.contribution.ts

```diff
@@ -0,0 +1,212 @@
+		@IExtensionManagementService readonly extensionManagementService: IExtensionManagementService,
+		@IActivityService readonly activityService: IActivityService,
+		@IExtensionService readonly extensionService: IExtensionService,
+		@IConfigurationService readonly configurationService: IConfigurationService,
+		@IContextKeyService readonly contextKeyService: IContextKeyService
+	) {
+		super();
+
+		if (!this.productService.entitlement) {
+			return;
+		}
+
+		// if previously shown, do not show again.
+		const showEntitlements = this.storageService.getBoolean(configurationKey, StorageScope.APPLICATION, true);
+		if (!showEntitlements) {
+			return;
+		}
+
+		const setting = this.configurationService.inspect<boolean>(configurationKey);
+		if (!setting.value) {
+			return;
+		}
+
+		this.extensionManagementService.getInstalled().then(exts => {
+			const installed = exts.find(value => ExtensionIdentifier.equals(value.identifier.id, this.productService.entitlement!.extensionId));
+			if (installed) {
+				this.storageService.store(configurationKey, false, StorageScope.APPLICATION, StorageTarget.MACHINE);
+				this.contextKey.set(false);
+				return;
+			} else {
+				this.registerListeners();
+			}
+		});
+	}
+
+	private registerListeners() {
+		this._register(this.extensionService.onDidChangeExtensions(async (result) => {
+			for (const ext of result.added) {
+				if (ExtensionIdentifier.equals(this.productService.entitlement!.extensionId, ext.identifier)) {
+					this.storageService.store(configurationKey, false, StorageScope.APPLICATION, StorageTarget.MACHINE);
+					this.contextKey.set(false);
+					return;
+				}
+			}
+		}));
+
+		this._register(this.authenticationService.onDidChangeSessions(async (e) => {
+			if (e.providerId === this.productService.entitlement!.providerId && e.event.added.length > 0 && !this.isInitialized) {
+				this.onSessionChange(e.event.added[0]);
+			} else if (e.providerId === this.productService.entitlement!.providerId && e.event.removed.length > 0) {
+				this.contextKey.set(false);
+			}
+		}));
+
+		this._register(this.authenticationService.onDidRegisterAuthenticationProvider(async e => {
+			if (e.id === this.productService.entitlement!.providerId && !this.isInitialized) {
+				const session = await this.authenticationService.getSessions(e.id);
+				this.onSessionChange(session[0]);
+			}
+		}));
+	}
+
+	private async onSessionChange(session: AuthenticationSession) {
+
+		this.isInitialized = true;
+
+		const init = {
+			method: 'GET',
+			headers: new Headers({
+				'Authorization': `Bearer ${session.accessToken}`
+			}),
+		};
+
+		const response = await fetch(this.productService.entitlement!.entitlementUrl, init);
+		if (!response.ok) {
+			return;
+		}
+		const parsedResult = await response.json();
+		if (!parsedResult) {
+			return;
+		}
+
+		const accountsMenuBadgeDisposable = this._register(new MutableDisposable());
+
+		const orgs = parsedResult['organization_login_list'] as any[];
```

Comment:
copilot specific stuff?

Reply:
Not really. Entitlements are tied to orgs, I decided to use it as is.